src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1853,7 +1853,7 @@ private bool ScanIdentifier_SlowPath(ref TokenInfo info)
                     Debug.Assert(string.Equals(info.Text.Substring(0, objectAddressOffset + 1), "@0x", StringComparison.OrdinalIgnoreCase));
                     var valueText = TextWindow.Intern(_identBuffer, objectAddressOffset, _identLen - objectAddressOffset);
                     // Verify valid hex value.
-                    if ((valueText.Length == 0) || !valueText.All(IsValidHexDigit))
+                    if ((valueText.Length == 0) || !valueText.All(s_isValidHexDigit))
```

Comment:
:memo: Jason pointed out that this only occurs for the special debugger syntax `@0xdeadf00d` (basically an address starting with `@0x`, where the `@` is not optional).